neural_compressor/torch/algorithms/base_algorithm.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+
+algo_quantizers = {}
+
+
+def algo_quantizer_register(name):
+    def decorator(algo_quantizer):
+        algo_quantizers[name] = algo_quantizer
+        return algo_quantizer
+
+    return decorator
+
+
+class BaseQuantizer(ABC):

```

Comment:
Suggest to follow this interface design. https://github.com/pytorch-labs/ao/blob/3124382b424beac3d7671beba0ea65a51daf137c/torchao/quantization/unified.py#L15